projects/packages/forms/src/service/class-mailpoet-integration.php

```diff
@@ -0,0 +1,191 @@
+			$new_list = $mailpoet_api->addList(
+				array(
+					'name'        => $list_name,
+					'description' => $list_description,
+				)
+			);
+			return $new_list['id'];
+		} catch ( \Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Add a subscriber to a MailPoet list.
+	 *
+	 * @param mixed  $mailpoet_api The MailPoet API instance.
+	 * @param string $list_id The MailPoet list ID.
+	 * @param array  $subscriber_data Associative array with at least 'email', optionally 'first_name', 'last_name'.
+	 * @return array|null Subscriber data on success, or null on failure.
+	 */
+	protected function add_subscriber_to_list( $mailpoet_api, $list_id, $subscriber_data ) {
+		try {
+			$subscriber = $mailpoet_api->addSubscriber(
+				$subscriber_data,
+				array( $list_id )
+			);
+			return $subscriber;
+		} catch ( \Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Extract subscriber data (email, first_name, last_name) from form fields.
+	 *
+	 * @param array $fields Collection of Contact_Form_Field instances.
+	 * @return array Associative array with at least 'email', optionally 'first_name', 'last_name'. Empty array if no email found.
+	 */
+	protected function get_subscriber_data_from_fields( $fields ) {
+		// Try and get the form from any of the fields
+		$form = null;
+		foreach ( $fields as $field ) {
+			if ( ! empty( $field->form ) ) {
+				$form = $field->form;
+				break;
+			}
+		}
+		if ( ! $form || ! is_a( $form, 'Automattic\Jetpack\Forms\ContactForm\Contact_Form' ) ) {
+			return array();
+		}
+
+		// Extract email, first_name, last_name from form fields.
+		$subscriber_data = array();
+		foreach ( $form->fields as $field ) {
+			$id    = strtolower( str_replace( array( ' ', '_' ), '', $field->get_attribute( 'id' ) ) );
+			$label = strtolower( str_replace( array( ' ', '_' ), '', $field->get_attribute( 'label' ) ) );
+			$value = trim( $field->value );
+
+			if ( ( $id === 'email' || $label === 'email' ) && ! empty( $value ) ) {
+				$subscriber_data['email'] = $value;
+			} elseif ( ( $id === 'firstname' || $label === 'firstname' ) && ! empty( $value ) ) {
+				$subscriber_data['first_name'] = $value;
+			} elseif ( ( $id === 'lastname' || $label === 'lastname' ) && ! empty( $value ) ) {
+				$subscriber_data['last_name'] = $value;
+			}
+		}
+
+		// Only return the subscriber data if we have an email.
+		if ( empty( $subscriber_data['email'] ) ) {
+			return array();
+		}
+
+		return $subscriber_data;
+	}
+
+	/**
+	 * Handle MailPoet integration after feedback post is inserted.
+	 *
+	 * @param int   $post_id      The post ID for the feedback CPT.
+	 * @param array $fields       Collection of Contact_Form_Field instances.
+	 * @param bool  $is_spam      Whether the submission is spam.
+	 * @param array $entry_values Extra fields from the contact form.
+	 */
+	public function handle_mailpoet_integration( $post_id, $fields, $is_spam, $entry_values ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
Lets leave the $entry_values out since we are not using them.

You will need to update to
`add_action( 'grunion_after_feedback_post_inserted', array( $this, 'handle_mailpoet_integration' ), 15, 3 );`

Reply:
Done!